plugins/nvme/nvme_test.py

```diff
@@ -0,0 +1,382 @@
+A: power/wakeup=disabled
+A: power/wakeup_abort_count=
+A: power/wakeup_active=
+A: power/wakeup_active_count=
+A: power/wakeup_count=
+A: power/wakeup_expire_count=
+A: power/wakeup_last_time_ms=
+A: power/wakeup_max_time_ms=
+A: power/wakeup_total_time_ms=
+A: power_state=D0
+A: reset_method=flr bus
+A: resource=0x0000000090a00000 0x0000000090a03fff 0x00000000001402040x0000000000000000 0x0000000000000000 0x00000000000000000x0000000000000000 0x0000000000000000 0x00000000000000000x0000000000000000 0x0000000000000000 0x00000000000000000x0000000000000000 0x0000000000000000 0x00000000000000000x0000000000000000 0x0000000000000000 0x00000000000000000x0000000000000000 0x0000000000000000 0x00000000000000000x0000000000000000 0x0000000000000000 0x00000000000000000x0000000000000000 0x0000000000000000 0x00000000000000000x0000000000000000 0x0000000000000000 0x00000000000000000x0000000000000000 0x0000000000000000 0x00000000000000000x0000000000000000 0x0000000000000000 0x00000000000000000x0000000000000000 0x0000000000000000 0x0000000000000000
+A: revision=0x01
+A: subsystem_device=0x5017
+A: subsystem_vendor=0x15b7
+A: vendor=0x15b7
+"""
+        )
+
+        self.testbed.add_from_string(
+            """P: /devices/pci0000:00/0000:00:02.4
+E: DRIVER=pcieport
+E: ID_PATH=pci-0000:00:02.4
+E: ID_PATH_TAG=pci-0000_00_02_4
+E: ID_PCI_CLASS_FROM_DATABASE=Bridge
+E: ID_PCI_INTERFACE_FROM_DATABASE=Normal decode
+E: ID_PCI_SUBCLASS_FROM_DATABASE=PCI bridge
+E: ID_VENDOR_FROM_DATABASE=Advanced Micro Devices, Inc. [AMD]
+E: MODALIAS=pci:v00001022d000014EEsv00001022sd00001453bc06sc04i00
+E: PCI_CLASS=60400
+E: PCI_ID=1022:14EE
+E: PCI_SLOT_NAME=0000:00:02.4
+E: PCI_SUBSYS_ID=1022:1453
+E: SUBSYSTEM=pci
+A: ari_enabled=0
+A: broken_parity_status=0
+A: class=0x060400
+H: config=2210EE14070410000000040610008100000000000000000000020200F1010000A090A090F1FF01000000000000000000000000005000000000000000FF000200
+A: consistent_dma_mask_bits=32
+A: current_link_speed=16.0 GT/s PCIe
+A: current_link_width=4
+A: d3cold_allowed=1
+A: device=0x14ee
+A: dma_mask_bits=32
+L: driver=../../../bus/pci/drivers/pcieport
+A: driver_override=(null)
+A: enable=1
+L: firmware_node=../../LNXSYSTM:00/LNXSYBUS:00/PNP0A08:00/device:0e
+L: iommu=../0000:00:00.2/iommu/ivhd0
+L: iommu_group=../../../kernel/iommu_groups/3
+A: irq=39
+A: local_cpulist=0-11
+A: local_cpus=0fff
+A: max_link_speed=16.0 GT/s PCIe
+A: max_link_width=4
+A: modalias=pci:v00001022d000014EEsv00001022sd00001453bc06sc04i00
+A: msi_bus=1
+A: msi_irqs/39=msi
+A: numa_node=-1
+A: power/autosuspend_delay_ms=100
+A: power/control=auto
+A: power/runtime_active_time=17184970
+A: power/runtime_status=active
+A: power/runtime_suspended_time=0
+A: power/wakeup=disabled
+A: power/wakeup_abort_count=
+A: power/wakeup_active=
+A: power/wakeup_active_count=
+A: power/wakeup_count=
+A: power/wakeup_expire_count=
+A: power/wakeup_last_time_ms=
+A: power/wakeup_max_time_ms=
+A: power/wakeup_total_time_ms=
+A: power_state=D0
```

Comment:
do we need all of these? I know it's easier to just dump everything, but for dozens of plugins it's going to be quite verbose.

Reply:
We don't need everything, I can pare down.